samples/aspnetcore/mvc/produces/src/ProducesMatcherPolicy/ProducesMatcherPolicy.cs

```diff
@@ -0,0 +1,256 @@
+            {
+                if (ordered[i].mediaType.MatchesAllTypes)
+                {
+                    exitDestination = ordered[i].destination;
+                    break;
+                }
+            }
+
+            return new ConsumesPolicyJumpTable(exitDestination, ordered);
+        }
+
+        private int GetScore(in MediaType mediaType)
+        {
+            // Higher score == lower priority - see comments on MediaType.
+            if (mediaType.MatchesAllTypes)
+            {
+                return 4;
+            }
+            else if (mediaType.MatchesAllSubTypes)
+            {
+                return 3;
+            }
+            else if (mediaType.MatchesAllSubTypesWithoutSuffix)
+            {
+                return 2;
+            }
+            else
+            {
+                return 1;
+            }
+        }
+
+        private class ProducesAttributeEndpointComparer : EndpointMetadataComparer<ProducesAttribute>
+        {
+            protected override int CompareMetadata(ProducesAttribute x, ProducesAttribute y)
+            {
+                // Ignore the metadata if it has an empty list of content types.
+                return base.CompareMetadata(
+                    x?.ContentTypes.Count > 0 ? x : null,
+                    y?.ContentTypes.Count > 0 ? y : null);
+            }
+        }
+
+        private class ConsumesPolicyJumpTable : PolicyJumpTable
+        {
+            private (MediaType mediaType, int destination)[] _destinations;
+            private int _exitDestination;
+
+            public ConsumesPolicyJumpTable(int exitDestination, (MediaType mediaType, int destination)[] destinations)
+            {
+                _exitDestination = exitDestination;
+                _destinations = destinations;
+            }
+
+            public override int GetDestination(HttpContext httpContext)
+            {
+                var acceptContentType = httpContext.Request.Headers[HeaderNames.Accept];
+                if (string.IsNullOrEmpty(acceptContentType))
+                {
+                    return _exitDestination;
+                }
+
+                var mediaTypesWithQuality = new List<MediaTypeSegmentWithQuality>();
+                AcceptHeaderParser.ParseAcceptHeader(acceptContentType, mediaTypesWithQuality);
+                mediaTypesWithQuality.Sort(_sortFunction);
+
+                var destinations = _destinations;
+
+                // Loop through accept values. Highest quality come first
+                for (int i = 0; i < mediaTypesWithQuality.Count; i++)
+                {
+                    var requestMediaType = new MediaType(mediaTypesWithQuality[i].MediaType);
+
+                    for (var j = 0; j < destinations.Length; j++)
+                    {
+                        if (requestMediaType.IsSubsetOf(destinations[j].mediaType)
+                            && !destinations[j].mediaType.MatchesAllTypes)
```

Comment:
I'm not sure about this. Should */* be in this collection? _exitDestination will head to */* anyway if there is no more exact match.

Reply:
How does rejection work in your approach? I thought I put everything *valid* in edges and used the exit for rejection.